pkg/util/modules/modules_test.go

```diff
@@ -0,0 +1,49 @@
+package modules
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+func mockInitFunc() (services.Service, error) { return nil, nil }
+
+func TestDependencies(t *testing.T) {
+	var testModules = map[string]module{
+		"serviceA": {
+			initFn: mockInitFunc,
+		},
+
+		"serviceB": {
+			initFn: mockInitFunc,
+		},
+
+		"serviceC": {
+			initFn: mockInitFunc,
+		},
+	}
+
+	mm := NewManager()
+	for name, mod := range testModules {
+		mm.RegisterModule(name, mod.initFn)
+	}
+	assert.NoError(t, mm.AddDependency("serviceB", "serviceA"))
+	assert.NoError(t, mm.AddDependency("serviceC", "serviceB"))
+	assert.Equal(t, mm.modules["serviceB"].deps, []string{"serviceA"})
+
+	invDeps := mm.findInverseDependencies("serviceA", []string{"serviceB", "serviceC"})
+	require.Len(t, invDeps, 1)
+	assert.Equal(t, invDeps[0], "serviceB")
+
+	svcs, err := mm.InitModuleServices("serviceC")
+	assert.NotNil(t, svcs)
+	assert.NoError(t, err)
+
+	svcs, err = mm.InitModuleServices("service_unknown")
```

Comment:
This test is interesting because it calls `InitModuleServices` twice... it's not clear from documentation what should actually happen in that case!

Would you mind adding a test for missing dependency too?